test/validation/api/timer/timer.c

```diff
@@ -145,18 +135,62 @@ static int timer_global_init(odp_instance_t *inst)
 		return -1;
 	}
 
-	global_mem = odp_shm_addr(global_shm);
-	memset(global_mem, 0, sizeof(global_shared_mem_t));
-
 	/* Configure scheduler */
 	odp_schedule_config(NULL);
 
+	return 0;
+}
+
+static int global_term(void)
+{
+	odp_shm_t shm;
+
+	shm = odp_shm_lookup(GLOBAL_SHM_NAME);
+	if (0 != odp_shm_free(shm)) {
```

Comment:
SHM handle could be saved into global mem and pass pointer here to avoid lookup.